src/session.ts

```diff
@@ -73,10 +74,11 @@ export abstract class BaseChannel implements Channel {
     }
 }
 export abstract class BaseSession implements Session {
-    t7: object
+    t7: Terminal7
     watchdog: number
-    onStateChange : (state: string, failure?: Failure) => void
+    onStateChange : (state: State, failure?: Failure) => void
     onPayloadUpdate: (payload: string) => void
+    isSSH: boolean
```

Comment:
This made the web exec install message never show. isSSH is defined int eh interface and every concrete class has a get function except for the base session. If your env is complaining and a get property function and raise an "Not Implemented" error.

Reply:
The problem isn't here. Look in `gate.ts:280`, the install message is commented out. I think we should move the entire `firstConnection()` to the shell and just handle it there.

Reply:
still, I don't think we should define `isSSH` here

Reply:
This is an abstract base class. As such, it's a full implementation of the interface. Some implementations are left to the concrete classes where each adds its own getter.

If we remove this line the implementation is left to the concrete class. If in the future we'll have a  class with no `isSSH` it will be undefined which is falsy. It's best not to let this go unnoticed so it's better to add a getter function that throws an error and lay a trap for the malformed classes of the future.

Reply:
Even if we keep this line it will basically mean nothing, as the property will stay undefined in inheriting classes (we're not assigning a value here).
Anyway I'll add a getter and throw an error there.